tests/test_dirs.py

```diff
@@ -360,6 +360,41 @@ def test_rootbuilddir_for_all_platforms_copy_invalid_file_path(
         root.for_all_platforms_copy(creation_func)
 
 
+def test_rootbuilddir_for_all_platforms_copy_dir_with_broken_symlinks(build_dir, mock_source):
+    """
+    Test that broken symlinks inside a directory are not followed/resolved when
+    copying build_dir contents to platforms dir.
+    """
+
+    def create_dir_with_broken_symlinks_inside(build_dir: BuildDir) -> Iterable[Path]:
+        directory = build_dir.path.joinpath("dir")
+        directory.mkdir()
+
+        symlink_to_nonexistent_file = directory.joinpath("symlink.file")
+        symlink_to_nonexistent_file.symlink_to(build_dir.path / directory / "nonexistent.file")
+
+        circular_symlink_1 = directory.joinpath("circular-symlink_1.file")
+        circular_symlink_2 = directory.joinpath("circular-symlink_2.file")
+
+        circular_symlink_1.symlink_to(circular_symlink_2)
+        circular_symlink_2.symlink_to(circular_symlink_1)
+
+        return [directory]
+
+    root = RootBuildDir(build_dir)
+    platforms = ["x86_64", "ppc64le"]
+    root.init_build_dirs(platforms, mock_source)
+    root.for_all_platforms_copy(create_dir_with_broken_symlinks_inside)
```

Comment:
why don't you just create all those files/dirs from the method after root.init_build_dirs ONCE, and then in the method just return build_dir.path.joinpath("dir") ????

all  you need is to copy directory from build_dir to all platforms